apps/babelstream/package.py

```diff
@@ -0,0 +1,268 @@
+        #             OMP 
+        #===================================
+        if '+omp' in spec:
+            if("cuda_arch" in spec.variants):
+                cuda_arch_list = spec.variants['cuda_arch'].value
+                # the architecture value is only number so append sm_ to the name
+                cuda_arch = "sm_" + cuda_arch_list[0]
+                args.append("-DOFFLOAD=ON" + " NVIDIA:" + cuda_arch)
+            elif("amdgpu_target" in spec.variants):
+                rocm_arch = spec.variants['amdgpu_target'].value
+                # the architecture value is only number so append sm_ to the name
+                args.append("-DOFFLOAD=ON" +" AMD:"+ rocm_arch)
+            # else:
+                # args.append("-DOFFLOAD=" +"INTEL")
+                
+
+
+        #===================================
+        #             SYCL 
+        #===================================
+       
+        if "+sycl" in spec:
+            # args.append("-DCXX_EXTRA_FLAGS= -fsycl -O3")
+            #this is required to enable -DCMAKE_CXX_COMPILER=icpx flag from CMake
+            args.append("-DSYCL_COMPILER=" + spec.variants['compiler'].value)
+
+
+        #===================================
+        #             SYCL 2020 
+        #===================================
+        
+        #cmake -Bbuild -H. -DMODEL=sycl2020 -DSYCL_COMPILER=ONEAPI-ICPX -DSYCL_COMPILER_DIR=ONEAPI-ICPX
+        if "+sycl2020" in spec:
+            # -fsycl flag is required for setting up sycl/sycl.hpp seems like it doesn't get it from the CMake file
+            args.append("-DCXX_EXTRA_FLAGS= -fsycl -O3")
+            #this is required to enable -DCMAKE_CXX_COMPILER=icpx flag from CMake
+            args.append("-DSYCL_COMPILER=ONEAPI-ICPX")
+
+        #===================================
+        #             HIP(ROCM) 
+        #===================================
+        
+        if "+rocm" in spec:
+            hip_comp = self.spec["rocm"].prefix + "/bin/hipcc"
+            args.append("-DCMAKE_CXX_COMPILER=" + hip_comp)
+            
+            rocm_archs = spec.variants['amdgpu_target'].value
+            if 'none' not in rocm_archs:
+                args.append(
+                    "-DCXX_EXTRA_FLAGS= --offload-arch="
+                    + self.spec.variants["amdgpu_target"].value
+                    + " "
+                    + self.spec.variants["flags"].value
+                    + " -O3"
+                )
+
+        #===================================
+        #             TBB 
+        #===================================
+
+        if "+tbb" in spec:
+            args.append("-DTBB_ROOT=" + self.spec["tbb"].prefix + "/lib/cmake/tbb")
+            # Hack to get around the fact that TBBConfig.cmake is linking 32bit libs.
+            args.append("-DCMAKE_SIZEOF_VOID_P=8")
+
+        #===================================
+        #             OpenCL (ocl) 
+        #===================================
+        if "+ocl" in spec:
```

Comment:
You might need to replicate some of the logic from CUDA to populate the `OpenCL_LIBRARY` variable with the right path based on if the OpenCL library is CUDA, HIP, Intel or POCL.